include/alpaka/core/CudaHipMath.hpp

```diff
@@ -15,6 +15,6 @@
 #    include <cuda_runtime.h>
 #endif
 
-#if defined(ALPAKA_ACC_GPU_HIP_ENABLED)
+#if defined(ALPAKA_ACC_GPU_HIP_ENABLED) && !defined(ALPAKA_HOST_ONLY)
```

Comment:
IMO instead of `!defined(ALPAKA_HOST_ONLY)`  it should be `defined(__HIP_DEVICE_COMPILE__)` to include hip math functions only within the device path.

In cupla https://github.com/alpaka-group/cupla/blob/287deacef75aac0833307e3d8aedabf42e4591ec/include/cupla/defines.hpp#L130  detecting the device compile is using an additional variable but maybe it is working with `__HIP_EVICE_COMPILE__` only.
@fwyzard could you try if this is working in your case?

Reply:
Sure, I'll check if that works too.

Reply:
Yes, using `defined(__HIP_DEVICE_COMPILE__)` instead of `!defined(ALPAKA_HOST_ONLY)` works.

Shall I make the change ?